src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,64 @@
-export const PeopleFilters = () => {
+import React from 'react';
+import { useSearchParams, Link } from 'react-router-dom';
+import cn from 'classnames';
+import { SexFilter } from '../types';
+import { getSearchWith } from '../utils/getSearchWith';
+
+type Props = {};
+
+const sexParams = [
+  { sortBy: SexFilter.All },
+  { sortBy: SexFilter.Male },
+  { sortBy: SexFilter.Female },
+];
+
+const centuriesParams = [
+  { sortBy: '16' },
+  { sortBy: '17' },
+  { sortBy: '18' },
+  { sortBy: '19' },
+  { sortBy: '20' },
+];
+
+export const PeopleFilters: React.FC<Props> = ({}) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || SexFilter.All;
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const handleQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const search = getSearchWith(
+      { query: e.target.value || null },
+      searchParams,
+    );
+
+    setSearchParams(search);
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
-          All
-        </a>
-        <a className="" href="#/people?sex=m">
-          Male
-        </a>
-        <a className="" href="#/people?sex=f">
-          Female
-        </a>
+        {sexParams.map((param, i) => {
+          return (
+            <Link
+              key={param.sortBy}
+              className={cn({
+                'is-active': sex === param.sortBy,
+              })}
+              to={{
+                search: getSearchWith(
+                  { sex: param.sortBy || null },
+                  searchParams,
+                ),
+              }}
+            >
+              {Object.keys(SexFilter)[i]}
+            </Link>
+          );
```

Comment:
It's not a good practice to use the index of an array as a key in a React component. React uses keys to identify which items have changed, are added, or are removed and it's not guaranteed that the index will be unique and consistent. Consider using a unique property from the data itself.